### DISCUSSION TITLE
Feedback - Fuel Istanbul Hacker House

### DISCUSSION THREAD

**Discussion by Dhaiwat10** [FuelLabs/fuels-ts#1492] [Ideas]
1. A ethers to fuels migration or comparison guide would be nice. Most of the hackers in the hacker house were new to Fuel but had _some_ previous experience with ethers or other Ethereum tooling.
2. I met some hackers who said that it would be easier for them to build React frontends with our SDK if we had dedicated React hooks. Something like [wagmi](https://wagmi.sh). Someone from the Fuel community already started this project - [fuels-react](https://github.com/vkzrx/fuels-react) but it is no longer maintained.
3. The docs for the wallet sdk and the ts sdk are not in the same place. This can be a bit tricky for new devs.
4. We should have something like [Rainbowkit](https://rainbowkit.com) for Fuel. Another React tooling request that is related to an Ethereum ecosystem counterpart. The frontend team has already built [something like this](https://github.com/FuelLabs/fuels-wallet/tree/master/packages/react) but, again, it's not as easy to use as Rainbowkit. This is something we should look at.
5. Our [official dev quickstart on the new Fuel Docs Hub](https://docs.fuel.network/guides/quickstart/) should utilize our new Fuels CLI and the scaffolding tool. Running typegen and build commands again and again is not needed anymore, and not a pleasant dev experience.
6. There are not enough (or _any_) written guides or video guides where we (or someone else from the community) showcases how to build an app start-to-finish using the SDK.
7. You need to have some balance in your wallet even if you want to just read from a contract (`.simulate()`). This is not ideal

**Comment by luizstacio** [comment]
Hey, this is nice feedback. I'm going to comment on some of them here:
2. We already have an implementation of the `hooks` for interacting with the SDK and Wallet SDK. `@fuel-wallet/react` you can take a look at the docs: https://docs.fuel.network/docs/wallet/dev/hooks-reference/ is getting better on the new version that is going to be released with the new Connectors.
3. This is a on going conversation. And with the new version of the Wallet SDK, we can move it to TS-SDK.
4. We recently implemented a new Connectors interface inside the Wallet SDK to create multiple connectors https://github.com/FuelLabs/fuels-wallet/pull/972. We are using the first version of it inside the https://alpha.fuel.network/bridge. This UI is provided by the package `@fuel-wallet/react`.
7. Not on the things we already solve, but it can be done using the `fundWithFakeUTXO` used by the `getTransactionCost`. We could add something similar on the dryRun method.

**Comment by nedsalk** [comment]
Related to point 1, there is already a similar issue #449. Mentioning it here for linking purposes.

**Comment by arboleya** [comment]
@Dhaiwat10 Could you please create individual tasks? _(just like I said [here](https://github.com/FuelLabs/fuels-ts/issues/1488#issuecomment-1846097598))_

**Comment by arboleya** [comment]
@Dhaiwat10 I converted this to a discussion.

We can open individual tasks as we go for the things we want to address.

**Comment by petertonysmith94** [comment]
@Dhaiwat10 Just found this 🥇 
Disclaimer: I was at the Fuel hacker house, but was very much learning the ropes.

I'd have personally found an architecture diagram really useful (see ChainLink docs below). For me there was a lot of new concepts and it's a big transition from Web2, so having a high level view of the ecosystem would have really helped get up to speed quicker 😃 

https://docs.chain.link/chainlink-functions/resources/architecture
![image](https://github.com/FuelLabs/fuels-ts/assets/16990131/d48731a5-b393-470c-b1ea-a2b9f7e3ce3a)

**Reply by arboleya** [reply]
Yes, a diagram can be very helpful.

> cc @FuelLabs/devrel

**Reply by petertonysmith94** [reply]
@arboleya
I was already doing one for my personally learning, happy to share once finished?
It would help validate my own learning as well - 2 birds, one stone

**Reply by sarahschwartz** [reply]
This is something the devrel team has on our roadmap ❤️

**Comment by Dhaiwat10** [comment]
Closing this discussion since most of the points already have an issue for them. The ones that don't fall outside the scope of the TS SDK in some shape or form and will be discussed with the devrel/frontend team accordingly. :)
